easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1000,38 +1049,23 @@ def quote_py_str(val):
 
 def to_template_str(value, templ_const, templ_val):
     """
-    Create string representation of provided value, using template values where possible.
-        - value can be a string, list, tuple, dict or combination thereof
+    Insert template values where possible
+        - value is a string
         - templ_const is a dictionary of template strings (constants)
         - templ_val is an ordered dictionary of template strings specific for this easyconfig file
     """
-    if isinstance(value, basestring):
-        # wrap string into quotes, except if it matches a template constant
-        if value not in templ_const.values():
-            value = quote_py_str(value)
-
-        old_value = None
-        while value != old_value:
-            old_value = value
-            if value in templ_const:
-                value = templ_const[value]
-            else:
-                # check for template values (note: templ_val dict is 'upside-down')
-                for tval, tname in templ_val.items():
-                    # only replace full words with templates: word to replace should be at the beginning of a line
-                    # or be preceded by a non-alphanumeric (\W). It should end at the end of a line or be succeeded
-                    # by another non-alphanumeric.
-                    value = re.sub(r'(^|\W)' + re.escape(tval) + r'(\W|$)', r'\1%(' + tname + r')s\2', value)
-    else:
```

Comment:
Yes, this method is now purely textual, the `value` parameter is a string, so we don't have to check for types anymore